ecosystem/sep-0048.md

```diff
@@ -817,6 +862,212 @@ For example, the `SCVal` in XDR, expressed in XDR-JSON:
+    SC_SPEC_EVENT_DATA_FORMAT_VEC = 1,
+    SC_SPEC_EVENT_DATA_FORMAT_MAP = 2
+};
+```
+
+##### Example
+
+In the Soroban Rust SDK the above structure describes an event such as:
+
+```rust
+/// Transfer event published when tokens are transferred.
+#[contractevent]
+pub struct Transfer {
+    /// The sender of the tokens.
+    #[topic] pub from: Address,
+    /// The recipient of the tokens.
+    #[topic] pub to: Address,
+    /// The amount transferred.
+    pub amount: i128,
+    /// The muxed identified of the recipient.
+    pub to_muxed_id: u64,
+}
+```
+
+The schema will be encoded to the following `SCSpecEntry` XDR when stored in
+the custom section. The `SCSpecEntry` expressed as XDR-JSON:
+
+```json
+{
+  "event_v0": {
+    "doc": "Transfer event published when tokens are transferred.",
+    "lib": "",
+    "name": "Transfer",
+    "prefixTopics": ["transfer"],
+    "params": [
+      {
+        "doc": "The sender of the tokens.",
+        "name": "from",
+        "type": "address",
+        "location": "topic_list"
+      },
+      {
+        "doc": "The recipient of the tokens.",
+        "name": "to",
+        "type": "address",
+        "location": "topic_list"
+      },
+      {
+        "doc": "The amount transferred.",
+        "name": "amount",
+        "type": "i128",
+        "location": "data"
+      },
+      {
+        "doc": "The muxed identifier of the recipient.",
+        "name": "to_muxed_id",
+        "type": "i128",
+        "location": "data"
+      }
+    ],
+    "data_format": "map"
+  }
+}
+```
+
+When used to publish an event:
+
+```rust
+Transfer {
+    from: Address::from_str("GB..."),
+    to: Address::from_str("GC..."),
+    amount: 123,
+    to_muxed_id: 456,
+}.publish(env);
+```
+
+It is equivalent to:
+
+```rust
+let from = Address::from_str(env, "GA...");
+let to = Address::from_str(env, "GB...");
+let amount = 123i128;
+let to_muxed_id = 456u64;
+env.events().publish(
+    // Topics:
+    (symbol_short!("transfer"), from, to),
+    // Data:
+    Map::<Symbol, Val>::from_array(
+        env,
+        [
+            (symbol_short!("amount"), amount.into_val(env)),
+            (Symbol::new(env, "to_muxed_id"), to_muxed_id.into_val(env)),
+        ],
+    ),
+);
+```
+
+And maps to the following `ContractEvent` XDR:
+
+```json
+{
+  "ext": "v0",
+  "contract_id": "CA...",
+  "type_": "contract",
+  "body": {
+    "v0": {
+      "topics": [
+        { "symbol": "transfer" },
+        { "address": "GB..." },
+        { "address": "GC..." }
+      ],
+      "data": {
+        "map": [
+          {
+            "key": { "symbol": "amount" },
+            "val": { "i128": "123" }
+          },
+          {
+            "key": { "symbol": "to_muxed_id" },
+            "val": { "u64": "456" }
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+Note: The Soroban Rust SDK, as demonstrated above, defaults to setting a single
+`prefixTopics` that is the snake_case transformation of the struct name. This
```

Comment:
@Shaptic snake_case, same as XDR-JSON, Horizon, etc, except RPC because RPC decided to go camelCase 😞 .

Reply:
lol I have no idea why we did that